src/gui/uithememanager.cpp

```diff
@@ -109,23 +110,25 @@ QIcon UIThemeManager::getIcon(const QString &iconId) const
 
 QIcon UIThemeManager::getIcon(const QString &iconId, const QString &fallback) const
 {
+    // cache to avoid rescaling svg icons
+    static QHash<QString, QIcon> iconCache;
+    const auto iter = iconCache.find(iconId);
+    if (iter != iconCache.end())
+        return *iter;
+
 #if (defined(Q_OS_UNIX) && !defined(Q_OS_MACOS))
     if (m_useSystemTheme) {
         QIcon icon = QIcon::fromTheme(iconId);
         if (icon.name() != iconId)
-            icon = QIcon::fromTheme(fallback, QIcon(IconProvider::instance()->getIconPath(iconId)));
+            icon = QIcon::fromTheme(fallback, QIcon(getIconPath(iconId)));
+        iconCache[iconId] = icon;
```

Comment:
Looks like you killed the system icon theme change handling in runtime...
I wouldn't introduce in this PR any changes except the main ones (i.e. change layout of qBittorrent icons).